deps/rabbitmq_auth_backend_oauth2/src/rabbit_auth_backend_oauth2.erl

```diff
@@ -298,15 +279,16 @@ post_process_payload_with_scope_alias_field_named(Payload, FieldName, ScopeAlias
        maps:put(?SCOPE_JWT_FIELD, ExpandedScopes, Payload).
 
 
--spec does_include_complex_claim_field(Payload :: map()) -> boolean().
-does_include_complex_claim_field(Payload) when is_map(Payload) ->
-        maps:is_key(application:get_env(?APP, ?COMPLEX_CLAIM_APP_ENV_KEY, undefined), Payload).
-
--spec post_process_payload_with_complex_claim(Payload :: map()) -> map().
-post_process_payload_with_complex_claim(Payload) ->
-    ComplexClaim = maps:get(application:get_env(?APP, ?COMPLEX_CLAIM_APP_ENV_KEY, undefined), Payload),
-    ResourceServerId = rabbit_data_coercion:to_binary(application:get_env(?APP, ?RESOURCE_SERVER_ID, <<>>)),
+-spec does_include_complex_claim_field(ResourceServerId :: binary(), Payload :: map()) -> boolean().
+does_include_complex_claim_field(ResourceServerId, Payload) when is_map(Payload) ->
+  case rabbit_oauth2_config:has_additional_scopes_key(ResourceServerId) of
```

Comment:
if `get_additonal_scopes_key/1` returned `{ok, Key} | {error, not_found}` there would be no reason to have the `has_additional_scope_key/1` function.